### PR TITLE
[Davranbek Jenisbaev]: Swagger setup and api docs for user crud

### PR DESCRIPTION
**Pull request by purple-mountain**

### CODE REVIEW

File: src/modules/users/decorators/swagger.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { applyDecorators } from '@nestjs/common';
```
Comment (Ismoil-JS) [inline src/modules/users/decorators/swagger.decorator.ts:1]: create file for every decorator, not in one file,

File: src/main.ts
Comment (Ismoil-JS) [inline src/main.ts:1]: Optional: try to use the index.ts file in every directory and export everything on this index.ts. You can create path, example, https://dev.to/rubiin/resolving-path-alias-in-nestjs-projects-11o1

Comment (purple-mountain) [reply src/main.ts:1]: I agree. But I try to create index.ts only when there are multiple files in one directory that are exported.

File: src/modules/users/decorators/swagger.decorator.ts (hunk 1 above)
Comment (purple-mountain) [reply src/modules/users/decorators/swagger.decorator.ts:1]: I also thought about it at first but then I decided it is better to have it in one place like we keep functionality of controller and service in one file. Also they all serve one purpose which is adding swagger documentation to users controller. Right now each decorator is very small and personally I think it is more readable and convenient to have them in one file.

File: src/main.ts
Comment (mukhammadjon-ss) [reply src/main.ts:1]: Both ways are correct. There is no any performance or security issue with that. Do what you prefer to do.

**Review by mukhammadjon-ss — APPROVED** [review]

**Review by farrooh — APPROVED** [review]

**Pull request merged**